test/test_util.py

```diff
@@ -582,6 +583,51 @@ def csv_validate_required_fields(self, config, callback):
                     # Allow client to assert on result.
                     callback(index, invalid_fields)
 
+    @with_context
+    def csv_validate_required_fields_case_insensitive(self):
+        """Test validate_required_fields against csv data
+        with DATA_SOURCE_ID, DATA_OWNER in uppercase."""
+        config = {'TASK_REQUIRED_FIELDS': {
+            'data_access': {'val': None, 'check_val': False},
+            'data_owner': {'val': None, 'check_val': False},
+            'data_source_id': {'val': None, 'check_val': False}}}
+
+        with patch.dict(self.flask_app.config, config):
+            cs = BulkTaskCSVImport(None)
+
+            # Check upper-case required fields.
+            cs._headers = ['sentence', 'DATA_ACCESS', 'DATA_SOURCE_ID', 'DATA_OWNER']
+            cs._check_required_headers()
+
+            # Check lower-case required fields.
+            cs._headers = ['sentence', 'data_access', 'data_source_id', 'data_owner']
+            cs._check_required_headers()
+
+            # Check mixed-case required fields.
+            cs._headers = ['sentence', 'Data_Access', 'Data_Source_Id', 'Data_Owner']
+            cs._check_required_headers()
+
+            # Check missing required field DATA_SOURCE_ID. Verify exception is raised.
+            cs._headers = ['sentence', 'data_access', 'Data_Owner']
+            try:
+                cs._check_required_headers()
+            except:
+                pass
```

Comment:
BulkImportException is not raised in this case. It is a Web runtime exception due to the context of the unit test running from within the Flask application. This was the reason for included `Exception` instead, which still handles the case of verifying an expected exception from validation. :)

Reply:
If you use request_context, it would work. The code changes would be

```python
from pybossa.importers import BulkImportException
...
    @with_request_context
    def csv_validate_required_fields_case_insensitive(self):
            ...
            # Check missing required field DATA_SOURCE_ID. Verify exception is raised.
            cs._headers = ['sentence', 'data_access', 'Data_Owner']
            assert_raises(BulkImportException, cs._check_required_headers)    
```

Reply:
Thank you! Fixed via https://github.com/bloomberg/pybossa/pull/736/commits/4b22374d2b49a174b8f44894b1cee9a7768ee69f